File: luxe-leather-nextjs/upload_login.py
import os
import requests
from dotenv import load_dotenv

load_dotenv('.env.local')

supabase_url = os.environ.get('NEXT_PUBLIC_SUPABASE_URL')
supabase_key = os.environ.get('SUPABASE_SERVICE_ROLE_KEY') or os.environ.get('NEXT_PUBLIC_SUPABASE_ANON_KEY')

file_path = '/Users/muhammadirtiza/.gemini/antigravity-ide/brain/f30b9ef5-38b0-4a65-af01-97fb15314806/login_hero_image_1782046110081.png'
target_path = 'platform-images/login_hero_image_new.png'

url = f"{supabase_url}/storage/v1/object/media/{target_path}"

headers = {
    "Authorization": f"Bearer {supabase_key}",
    "Content-Type": "image/png"
}

with open(file_path, 'rb') as f:
    data = f.read()

print("Uploading...")
response = requests.post(url, headers=headers, data=data)

if response.status_code in [200, 201]:
    print("Success:", response.json())
else:
    print("Error:", response.status_code, response.text)

File: luxe-leather-nextjs/upload_login.mjs
import { execSync } from 'child_process';
import { createClient } from '@supabase/supabase-js';
import dotenv from 'dotenv';
dotenv.config({ path: '.env.local' });

const supabaseUrl = process.env.NEXT_PUBLIC_SUPABASE_URL;
const supabaseKey = process.env.SUPABASE_SERVICE_ROLE_KEY || process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY;
const supabase = createClient(supabaseUrl, supabaseKey);

async function main() {
    const file = '/Users/muhammadirtiza/.gemini/antigravity-ide/brain/f30b9ef5-38b0-4a65-af01-97fb15314806/login_hero_image_1782046110081.png';
    const filePath = 'platform-images/login_hero_image_new.png';

    const curlCmd = `curl -X POST "${supabaseUrl}/storage/v1/object/media/${filePath}" \\
    -H "Authorization: Bearer ${supabaseKey}" \\
    -H "Content-Type: image/png" \\
    --data-binary @"${file}"`;

    try {
        console.log("Uploading via curl...");
        execSync(curlCmd, { stdio: 'inherit' });
        console.log("\\nUploaded via curl.");
        const url = supabase.storage.from('media').getPublicUrl(filePath).data.publicUrl;
        
        const { data: existing } = await supabase.from('site_settings').select('*').eq('key', 'login_hero_image').single();
        if (existing) {
            await supabase.from('site_settings').update({ value: url }).eq('key', 'login_hero_image');
        } else {
            await supabase.from('site_settings').insert({ key: 'login_hero_image', value: url });
        }
        console.log(`Updated db for login_hero_image to ${url}`);
    } catch(e) {
        console.error("Upload failed", e);
    }
}
main();
